### DISCUSSION TITLE
Windows build

### DISCUSSION THREAD

**Discussion by VytPal** [trustwallet/wallet-core#4060] [Q&A]
I’m working on building Trust Wallet Core for a Windows desktop application and could use some guidance on the steps or any necessary changes to make the project compatible with the Windows platform.

I’ve come across the [kaetemi/wallet-core-windows](https://github.com/kaetemi/wallet-core-windows) port, but it hasn’t been updated in over 3 years, so I’m concerned it might be too outdated for current use.

Could anyone provide insight into how to properly configure and build Trust Wallet Core for Windows?

**Comment by satoshiotomakan** [comment]
Hi @VytPal, unfortunately, we don't support compiling WalletCore on Windows. Do you need C++ bindings library built on Windows?
On the other hands, you should be able to easily compile Rust crates within `./rust` directory by running `cargo build`. Hope this will help you.
Should you have any question, feel free to ask

**Reply by VytPal** [reply]
Hi @satoshiotomakan, 
Couple questions: 
1. Is there a list of chains currently supported on Rust?
2. Does Rust and C++ use the same C headers?

**Reply by satoshiotomakan** [reply]
Hi @VytPal, please run to get number of Rust supported chains.
```
cargo run --manifest-path rust/Cargo.toml -p wallet_core_bin -- registry-stats
```

No, the WalletCore layer is as follows. Rust FFI headers are generated automatically, you can find them at: `./src/rust/bindgen/WalletCoreRSBindgen.h`, but C++ headers are at `./include/`

                                FFI                           FFI
                                  v                              v
Kotlin/Swift/TS API -> C++ code base -> Rust code base

> Does Rust and C++ use the same C headers?
